### PR TITLE
Only setup nix-related tmpfiles if nix is enabled

### PR DESCRIPTION
**Pull request by jmbaur**
## Description of changes

If nix is not installed on the end system, the tmpfiles settings related to it do not need to be setup.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jmbaur** [comment]
@ofborg test nix-upgrade

**Review by r-vdp — APPROVED** [review]
LGTM!

File: nixos/modules/system/activation/activation-script.nix [hunk 1]
```diff
@@ -233,10 +233,11 @@ in
     system.activationScripts.stdio = ""; # obsolete
     system.activationScripts.var = ""; # obsolete
 
-    systemd.tmpfiles.rules = [
+    systemd.tmpfiles.rules = lib.optionals config.nix.enable [
```
Comment (SuperSandro2000) [inline nixos/modules/system/activation/activation-script.nix:236]: We should swap the none conditional files to be in the front

Comment (jmbaur) [reply nixos/modules/system/activation/activation-script.nix]: I reordered them to have these last

File: nixos/modules/system/boot/systemd/tmpfiles.nix [hunk 2]
```diff
@@ -281,15 +281,19 @@ in
...
+    ]
+    # Boot-time cleanup
+    ++ [
       "R! /etc/group.lock                    -    -    -    - -"
       "R! /etc/passwd.lock                   -    -    -    - -"
       "R! /etc/shadow.lock                   -    -    -    - -"
+    ] ++ lib.optionals config.nix.enable [
       "R! /nix/var/nix/gcroots/tmp           -    -    -    - -"
       "R! /nix/var/nix/temproots             -    -    -    - -"
     ];
```
Comment (SuperSandro2000) [inline nixos/modules/system/boot/systemd/tmpfiles.nix:299]:
```suggestion
    # Boot-time cleanup
      "R! /etc/group.lock                    -    -    -    - -"
      "R! /etc/passwd.lock                   -    -    -    - -"
      "R! /etc/shadow.lock                   -    -    -    - -"
    ] ++ lib.optionals config.nix.enable [
      "d  /nix/var                           0755 root root - -"
      "L+ /nix/var/nix/gcroots/booted-system 0755 root root - /run/booted-system"
      "R! /nix/var/nix/gcroots/tmp           -    -    -    - -"
      "R! /nix/var/nix/temproots             -    -    -    - -"
    ];
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

File: nixos/modules/system/boot/systemd/tmpfiles.nix (hunk 2 above)
Comment (jmbaur) [reply nixos/modules/system/boot/systemd/tmpfiles.nix:299]: I don't know about this, the tmpfiles rules are ordered by the tmpfiles specifier (`d`, `L`, etc.) and the comments are no longer accurate if we change this ordering. I'd prefer to keep it as it is now.

**Comment by jmbaur** [comment]
> Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

What would you prefer the commit titles to be? I believe they already follow the guidelines.

**Comment by r-vdp** [comment]
> Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

Are you confusing the PR title for the commit messages?
AFAICT, the actual commit messages are fine.

File: nixos/modules/system/boot/systemd/tmpfiles.nix (hunk 2 above)
Comment (ElvishJerricco) [reply nixos/modules/system/boot/systemd/tmpfiles.nix:299]: Yea, that reordering would be a functional change. It's fine as is.

Comment (ElvishJerricco) [reply nixos/modules/system/boot/systemd/tmpfiles.nix:299]: Yea, I agree. Such reordering is unnecessary and inappropriate. It's fine as is.

**Pull request merged**